src/test/java/org/elasticsearch/index/query/MatchAllQueryBuilderTest.java

```diff
@@ -0,0 +1,174 @@
+                new SimilarityModule(settings),
+                new IndexNameModule(index),
+                new IndexQueryParserModule(settings),
+                new FunctionScoreModule(),
+                new AbstractModule() {
+                    @Override
+                    protected void configure() {
+                        bind(ClusterService.class).toProvider(Providers.of((ClusterService) null));
+                        bind(CircuitBreakerService.class).to(NoneCircuitBreakerService.class);
+                    }
+                }
+        ).createInjector();
+
+        IndexQueryParserService queryParserService = injector.getInstance(IndexQueryParserService.class);
+        context = new QueryParseContext(index, queryParserService);
+
+        testQuery = createTestQuery();
+        String contentString = createXContent(testQuery).string();
+        parser = XContentFactory.xContent(contentString).createParser(contentString);
+    }
+
+    XContentBuilder createXContent(BaseQueryBuilder query) throws IOException {
+        XContentBuilder content = XContentFactory.jsonBuilder();
+        query.toXContent(content, null);
+        return content;
+    }
+
+    @Override
+    @After
+    public void tearDown() throws Exception {
+        super.tearDown();
+        terminate(injector.getInstance(ThreadPool.class));
+    }
+
+    @Test
+    public void testToXContent() throws IOException {
+        XContentBuilder content = createXContent(new MatchAllQueryBuilder().boost(1.5f));
+        assertEquals(content.string(), MATCH_ALL_BOOST_1_5);
+    }
+
+    @Test
+    public void testFromXContent() throws IOException {
+        context.reset(parser);
+        MatchAllQueryBuilder newMatchAllQuery = injector.getInstance(MatchAllQueryBuilder.class);
+        newMatchAllQuery.fromXContent(context);
+        // compare these
+        assertThat(testQuery.getBoost(), is(newMatchAllQuery.getBoost()));
+    }
+
+    @Test
+    public void testToQuery() throws IOException {
+        context.reset(parser);
+        MatchAllQueryBuilder newMatchAllQuery = injector.getInstance(MatchAllQueryBuilder.class);
+        newMatchAllQuery.fromXContent(context);
+        Query query = newMatchAllQuery.toQuery(context);
+        // compare these
+        assertThat(query.getBoost(), is(testQuery.getBoost()));
+    }
+
+    @Test
+    public void testSerialization() throws IOException {
+        context.reset(parser);
+
+        BytesStreamOutput output = new BytesStreamOutput();
+        testQuery.writeTo(output);
+
+        BytesStreamInput bytesStreamInput = new BytesStreamInput(output.bytes());
+        MatchAllQueryBuilder deserializedQuery = new MatchAllQueryBuilder();
+        deserializedQuery.readFrom(bytesStreamInput);
+
+        assertEquals(deserializedQuery, testQuery);
+        assertNotSame(deserializedQuery, testQuery);
+        assertThat(createXContent(testQuery).string(), is(createXContent(deserializedQuery).string()));
```

Comment:
I would probably take out the string comparison, once we know that the queries are equal, I think that's enough, we shouldn't test other methods on them, it's out of the scope of this test.